traversal/selector/matcher_util.go

```diff
@@ -0,0 +1,15 @@
+package selector
+
+import "io"
+
+type readerat struct {
+	io.ReadSeeker
```

Comment:
note that ReaderAt is safe for concurrent use and ReadSeeker isn't, so you should probably grab a sync.Mutex or some other form of semaphore on every ReadAt call

Reply:
Since i only use this within the readseeker that is non-concurrent again, is it okay to note that it isn't actually concurrent safe?